cmd2/cmd2.py

```diff
@@ -2176,120 +2245,355 @@ def _cmdloop(self) -> bool:
+                    self.poutput("Alias '{}' deleted".format(cur_name))
+                else:
+                    self.perror("Alias '{}' does not exist".format(cur_name), traceback_war=False)
+
+    def alias_list(self, args: argparse.Namespace):
+        """ Lists some or all aliases """
+        if args.name:
+            # Get rid of duplicates and strip quotes since the argparse decorator for do_alias() preserves them
+            names_to_view = [utils.strip_quotes(cur_name) for cur_name in utils.remove_duplicates(args.name)]
+
+            for cur_name in names_to_view:
+                if cur_name in self.aliases:
+                    self.poutput("alias create {} {}".format(cur_name, self.aliases[cur_name]))
+                else:
+                    self.perror("Alias '{}' not found".format(cur_name), traceback_war=False)
+        else:
+            sorted_aliases = utils.alphabetical_sort(self.aliases)
             for cur_alias in sorted_aliases:
-                self.poutput("alias {} {}".format(cur_alias, self.aliases[cur_alias]))
+                self.poutput("alias create {} {}".format(cur_alias, self.aliases[cur_alias]))
+
+    # Top-level parser for alias
+    alias_description = ("Manage aliases\n"
+                         "\n"
+                         "An alias is a command that enables replacement of a word by another string.")
+    alias_epilog = ("See also:\n"
+                    "  macro")
+    alias_parser = ACArgumentParser(description=alias_description, epilog=alias_epilog, prog='alias')
+
+    # Add subcommands to alias
+    alias_subparsers = alias_parser.add_subparsers()
+
+    # alias -> create
+    alias_create_help = "create or overwrite an alias"
+    alias_create_description = "Create or overwrite an alias"
+
+    alias_create_epilog = ("Notes:\n"
+                           "  If you want to use redirection or pipes in the alias, then quote them to\n"
+                           "  prevent the 'alias create' command from being redirected.\n"
+                           "\n"
+                           "  Since aliases are resolved during parsing, tab completion will function as it\n"
+                           "  would for the actual command the alias resolves to.\n"
+                           "\n"
+                           "Examples:\n"
+                           "  alias ls !ls -lF\n"
+                           "  alias create show_log !cat \"log file.txt\"\n"
+                           "  alias create save_results print_results \">\" out.txt\n")
+
+    alias_create_parser = alias_subparsers.add_parser('create', help=alias_create_help,
+                                                      description=alias_create_description,
+                                                      epilog=alias_create_epilog)
+    setattr(alias_create_parser.add_argument('name', help='name of this alias'),
+            ACTION_ARG_CHOICES, get_commands_and_aliases_for_completion)
```

Comment:
I suggest no tab completion for macro and alias names

Reply:
If we aren't already doing a check, I think we should not allow users to create aliases or macros with the same name as a command; otherwise this leads to potentially pathological situations like the one Lance mentioned.